### PR TITLE
Azure: Add the parser for azure-windows

### PR DESCRIPTION
**Pull request by squioc**
Add the parser for the ingestion of events for azure-windows.

Testing files come from the previous parser (with some additions to match new expectations) to validate the backward compatibility of this parser.

### CODE REVIEW

File: Azure/azure-windows/_meta/fields.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+###############################################################
+#                                                             #
+# next fields are declared for backward compatibility purpose #
+#                                                             #
+###############################################################
+
+action.properties:
+  name: action.properties
+  description: "A list of objects"
+  type: text
```
Comment (LeoArurault) [inline Azure/azure-windows/_meta/fields.yml:56]: Is the type coherent with values it should contains ?

File: Azure/azure-windows/ingest/parser.yml [hunk 2]
```diff
@@ -0,0 +1,764 @@
...
+    description: 'Parse the windows event in the message'
+    external:
+      name: xml.parse-windows-event
+      properties:
+        input_field: '{{parse_json.message.properties.RawXml}}'
+        output_field: message
+
+  - name: extract_provider_guid
+    description: 'Extract the Provider identifier from the GUID'
+    filter: ''
```
Comment (LeoArurault) [inline Azure/azure-windows/ingest/parser.yml:21]:
```suggestion
```

File: Azure/azure-windows/ingest/parser.yml [hunk 3]
```diff
@@ -0,0 +1,764 @@
...
+    external:
+      name: grok.match
+      properties:
+        input_field: '{{parse_json.message.properties.ProviderGuid}}'
+        output_field: result
+        pattern: '{%{UUID:provider_guid}}'
```
Comment (LeoArurault) [inline Azure/azure-windows/ingest/parser.yml:27]: Do we need a grok there ? It seems there is no data structure, so you could just use a `set`

File: Azure/azure-windows/tests/kerberos.json [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+                "domain": {"name": "ACME.LOCAL"},
+                "identifier": "S-1-5-21-1004336348-2052111302-725345543-33053",
+                "name": "HOSTMON",
+                "type": "targetedUser"
+            }
+        },
+        "related": {
+            "ip": [
+                "10.129.224.1"
+            ]
```
Comment (LeoArurault) [inline Azure/azure-windows/tests/kerberos.json:114]: No related user there ?

File: Azure/azure-windows/tests/Event_4719.json [hunk 5]
```diff
@@ -0,0 +1,92 @@
+{
+    "input": {
+        "event": {
+            "created": "2021-01-11T10:48:46.476330800Z",
+            "original": "1sjze813YtXlmgHp3a1jU4rOAwYpBKMFaWtYCeTQ0QhEtg36Z68bcNi4ahZ2G7Fz",
+            "dialect": "azure windows",
+            "dialect_uuid": "1e256ea1-3947-429e-97a6-abaec8702dc4",
+            "id": "10f0afe9-98a1-4226-a6bd-8f70d461d430",
+            "outcome": "failure"
+        },
```
Comment (LeoArurault) [inline Azure/azure-windows/tests/Event_4719.json:10]:
```suggestion
```

File: Azure/azure-windows/tests/kerberos.json [hunk 6]
```diff
@@ -0,0 +1,124 @@
+{
+    "input": {
+        "event": {
+            "created": "2019-07-22T11:20:54.558577600Z",
+            "original": "1sjze813YtXlmgHp3a1jU4rOAwYpBKMFaWtYCeTQ0QhEtg36Z68bcNi4ahZ2G7Fz",
+            "dialect": "azure windows",
+            "dialect_uuid": "1e256ea1-3947-429e-97a6-abaec8702dc4",
+            "id": "10f0afe9-98a1-4226-a6bd-8f70d461d430",
+            "outcome": "failure"
+        },
```
Comment (LeoArurault) [inline Azure/azure-windows/tests/kerberos.json:10]:
```suggestion
```

File: Azure/azure-windows/tests/key_file_operation.json [hunk 7]
```diff
@@ -0,0 +1,95 @@
+{
+    "input": {
+        "event": {
+            "created": "2019-06-24T09:20:18.054208500Z",
+            "original": "1sjze813YtXlmgHp3a1jU4rOAwYpBKMFaWtYCeTQ0QhEtg36Z68bcNi4ahZ2G7Fz",
+            "dialect": "azure windows",
+            "dialect_uuid": "1e256ea1-3947-429e-97a6-abaec8702dc4",
+            "id": "10f0afe9-98a1-4226-a6bd-8f70d461d430",
+            "outcome": "failure"
+        },
```
Comment (LeoArurault) [inline Azure/azure-windows/tests/key_file_operation.json:10]:
```suggestion
```

File: Azure/azure-windows/tests/loggedoff.json [hunk 8]
```diff
@@ -0,0 +1,88 @@
+{
+    "input": {
+        "event": {
+            "created": "2019-07-23T15:33:09.199351700Z",
+            "original": "1sjze813YtXlmgHp3a1jU4rOAwYpBKMFaWtYCeTQ0QhEtg36Z68bcNi4ahZ2G7Fz",
+            "dialect": "azure windows",
+            "dialect_uuid": "1e256ea1-3947-429e-97a6-abaec8702dc4",
+            "id": "10f0afe9-98a1-4226-a6bd-8f70d461d430",
+            "outcome": "failure"
+        },
```
Comment (LeoArurault) [inline Azure/azure-windows/tests/loggedoff.json:10]:
```suggestion
```

File: Azure/azure-windows/tests/process_operation.json [hunk 9]
```diff
@@ -0,0 +1,109 @@
+{
+    "input": {
+        "event": {
+            "created": "2019-07-16T14:16:10.209241300Z",
+            "original": "1sjze813YtXlmgHp3a1jU4rOAwYpBKMFaWtYCeTQ0QhEtg36Z68bcNi4ahZ2G7Fz",
+            "dialect": "azure windows",
+            "dialect_uuid": "1e256ea1-3947-429e-97a6-abaec8702dc4",
+            "id": "10f0afe9-98a1-4226-a6bd-8f70d461d430",
+            "outcome": "failure"
+        },
```
Comment (LeoArurault) [inline Azure/azure-windows/tests/process_operation.json:10]:
```suggestion
```

File: Azure/azure-windows/tests/process_operation2.json [hunk 10]
```diff
@@ -0,0 +1,99 @@
+{
+    "input": {
+        "event": {
+            "created": "2019-07-22T12:54:05.281641000Z",
+            "original": "1sjze813YtXlmgHp3a1jU4rOAwYpBKMFaWtYCeTQ0QhEtg36Z68bcNi4ahZ2G7Fz",
+            "dialect": "azure windows",
+            "dialect_uuid": "1e256ea1-3947-429e-97a6-abaec8702dc4",
+            "id": "10f0afe9-98a1-4226-a6bd-8f70d461d430",
+            "outcome": "failure"
+        },
```
Comment (LeoArurault) [inline Azure/azure-windows/tests/process_operation2.json:10]:
```suggestion
```

File: Azure/azure-windows/tests/sysmon.json [hunk 11]
```diff
@@ -0,0 +1,118 @@
+{
+    "input": {
+        "event": {
+            "created": "2019-08-30T14:53:03.064863900Z",
+            "original": "1sjze813YtXlmgHp3a1jU4rOAwYpBKMFaWtYCeTQ0QhEtg36Z68bcNi4ahZ2G7Fz",
+            "dialect": "azure windows",
+            "dialect_uuid": "1e256ea1-3947-429e-97a6-abaec8702dc4",
+            "id": "10f0afe9-98a1-4226-a6bd-8f70d461d430",
+            "outcome": "failure"
+        },
```
Comment (LeoArurault) [inline Azure/azure-windows/tests/sysmon.json:10]:
```suggestion
```

File: Azure/azure-windows/tests/sysmon_3.json [hunk 12]
```diff
@@ -0,0 +1,125 @@
+{
+    "input": {
+        "event": {
+            "created": "2019-12-18T16:57:17.936358800Z",
+            "original": "1sjze813YtXlmgHp3a1jU4rOAwYpBKMFaWtYCeTQ0QhEtg36Z68bcNi4ahZ2G7Fz",
+            "dialect": "azure windows",
+            "dialect_uuid": "1e256ea1-3947-429e-97a6-abaec8702dc4",
+            "id": "10f0afe9-98a1-4226-a6bd-8f70d461d430",
+            "outcome": "failure"
+        },
```
Comment (LeoArurault) [inline Azure/azure-windows/tests/sysmon_3.json:10]:
```suggestion
```

**Review by LeoArurault — CHANGES_REQUESTED** [review]

File: Azure/azure-windows/ingest/parser.yml (hunk 2 above)
Comment (squioc) [reply Azure/azure-windows/ingest/parser.yml]: Fixed

File: Azure/azure-windows/tests/kerberos.json (hunk 4 above)
Comment (squioc) [reply Azure/azure-windows/tests/kerberos.json]: Fixed

File: Azure/azure-windows/ingest/parser.yml (hunk 3 above)
Comment (squioc) [reply Azure/azure-windows/ingest/parser.yml]: I simplify the way to extract the provider guid from the '{}'

File: Azure/azure-windows/tests/working_dir.json [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+            "dialect": "azure windows",
+            "dialect_uuid": "2815eaab-2425-4eff-8038-3f7d5a3b8b11",
+            "id": "f6bce3b9-737e-410f-b807-a4d8048b4fde",
+            "outcome": "failure"
+        },
```
[the remark this answers is not in the document]
Comment (squioc) [reply Azure/azure-windows/tests/working_dir.json:10]: applied

File: Azure/azure-windows/_meta/fields.yml [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+action.target:
+  name: action.target
+  description: "The target of the action"
+  type: keyword
+
+process.ppid:
+  name: process.ppid
+  description: "The identifier of the parent of the process"
+  type: number
+
```
Comment (LeoArurault) [inline Azure/azure-windows/_meta/fields.yml:67]: Already exists in ECS https://www.elastic.co/guide/en/ecs/1.7/ecs-process.html#field-process-ppid
```suggestion
```

**Review by LeoArurault — CHANGES_REQUESTED** [review]
Almost good :muscle: 
To have cleaner tests, I think you could also remove `expected.sekoiaio` fields and part of `expected.event` (like dialect, created, provider, original; dialect_uuid, id)

File: Azure/azure-windows/_meta/fields.yml (hunk 14 above)
Comment (squioc) [reply Azure/azure-windows/_meta/fields.yml]: In the last versions of the specification, `process.ppid` was replaced by `process.parent.pid`.
I kept this field in the taxonomy for backward compatibility. However, after analyzing the detection rules, it seems that field is not used (so I removed it from the format).

**Review by LeoArurault — APPROVED** [review]
Let's try this on prod now :rocket:

**Pull request merged**